### ISSUE TITLE
[Bug] Jupyter-lab kernel dying when running segmentation.write_patch_cells

### ISSUE THREAD

**Issue by lguerard** [gustaveroussy/sopa#131]
## Description
Probably not _directly_ sopa related but I'm still baffled. For the last 2 days, everytime I run `segmentation.write_patch_cells`, I don't see a lot of CPU/RAM/disk action but my kernel dies in my notebook without any informative output. I tried using an older version of sopa and it also crashes. Everything was running smoothly before. 

Would you have any idea why ? I see that there have been some `spatialdata` releases in the meantime, could it be one of the dependency ? I tried on multiple machines with multiple versions of conda and jupyterlab, but it always fails :/

## System
 - OS: Windows 10
 - Python version 3.10
<details>
<summary>Dependencies versions</summary>
<br>

> Package                   Version
> ------------------------- --------------
> aiobotocore               2.11.2
> aiohappyeyeballs          2.4.0
> aiohttp                   3.10.6
> aioitertools              0.12.0
> aiosignal                 1.3.1
> anndata                   0.10.9
> anyio                     4.6.0
> argon2-cffi               23.1.0
> argon2-cffi-bindings      21.2.0
> array_api_compat          1.8
> arrow                     1.3.0
> asciitree                 0.3.3
> asttokens                 2.4.1
> async-lru                 2.0.4
> async-timeout             4.0.3
> attrs                     24.2.0
> babel                     2.16.0
> beautifulsoup4            4.12.3
> bleach                    6.1.0
> botocore                  1.34.19
> Brotli                    1.1.0
> cached-property           1.5.2
> cellpose                  3.0.11
> certifi                   2024.8.30
> cffi                      1.17.1
> charset-normalizer        3.3.2
> click                     8.1.7
> cloudpickle               3.0.0
> colorama                  0.4.6
> colorcet                  3.1.0
> comm                      0.2.2
> contourpy                 1.3.0
> cycler                    0.12.1
> dask                      2024.9.0
> dask-expr                 1.1.14
> dask-image                2024.5.3
> datashader                0.16.3
> datrie                    0.8.2
> debugpy                   1.8.6
> decorator                 5.1.1
> defusedxml                0.7.1
> dill                      0.3.8
> distributed               2024.9.0
> entrypoints               0.4
> exceptiongroup            1.2.2
> executing                 2.1.0
> fasteners                 0.19
> fastjsonschema            2.20.0
> fastremap                 1.15.0
> fcsparser                 0.2.8
> filelock                  3.16.1
> flake8                    7.1.1
> fonttools                 4.54.1
> fqdn                      1.5.1
> frozenlist                1.4.1
> fsspec                    2023.6.0
> geopandas                 1.0.1
> h11                       0.14.0
> h2                        4.1.0
> h5py                      3.11.0
> hpack                     4.0.0
> httpcore                  1.0.5
> httpx                     0.27.2
> hyperframe                6.0.1
> idna                      3.10
> imagecodecs               2024.9.22
> imageio                   2.35.1
> importlib_metadata        8.5.0
> importlib_resources       6.4.5
> ipykernel                 6.29.5
> ipython                   8.27.0
> ipywidgets                8.1.5
> isoduration               20.11.0
> jedi                      0.19.1
> Jinja2                    3.1.4
> jmespath                  1.0.1
> joblib                    1.4.2
> json5                     0.9.25
> jsonpointer               3.0.0
> jsonschema                4.23.0
> jsonschema-specifications 2023.12.1
> jupyter_client            8.6.3
> jupyter_core              5.7.2
> jupyter-events            0.10.0
> jupyter-lsp               2.2.5
> jupyter_server            2.14.2
> jupyter_server_terminals  0.5.3
> jupyterlab                4.2.5
> jupyterlab_pygments       0.3.0
> jupyterlab_server         2.27.3
> jupyterlab_widgets        3.0.13
> kiwisolver                1.4.7
> lamin_utils               0.13.4
> lazy_loader               0.4
> legacy-api-wrap           1.4
> llvmlite                  0.43.0
> locket                    1.0.0
> loompy                    3.0.7
> markdown-it-py            3.0.0
> MarkupSafe                2.1.5
> matplotlib                3.9.2
> matplotlib-inline         0.1.7
> mccabe                    0.7.0
> mdurl                     0.1.2
> mistune                   3.0.2
> mpmath                    1.3.0
> msgpack                   1.1.0
> multidict                 6.1.0
> multipledispatch          1.0.0
> multiprocess              0.70.16
> multiscale_spatial_image  1.0.1
> natsort                   8.4.0
> nbclient                  0.10.0
> nbconvert                 7.16.4
> nbformat                  5.10.4
> nest-asyncio              1.6.0
> networkx                  3.3
> notebook_shim             0.2.4
> numba                     0.60.0
> numcodecs                 0.13.0
> numpy                     1.26.4
> numpy-groupies            0.11.2
> ome-zarr                  0.9.0
> opencv-python             4.10.0.84
> opencv-python-headless    4.10.0.84
> overrides                 7.7.0
> p_tqdm                    1.4.2
> packaging                 24.1
> pandas                    2.2.3
> pandocfilters             1.5.1
> param                     2.1.1
> parso                     0.8.4
> partd                     1.4.2
> pathos                    0.3.2
> patsy                     0.5.6
> pickleshare               0.7.5
> pillow                    10.4.0
> PIMS                      0.7
> pip                       24.2
> pkgutil_resolve_name      1.3.10
> platformdirs              4.3.6
> pooch                     1.8.2
> pox                       0.3.4
> ppft                      1.7.6.8
> prometheus_client         0.21.0
> prompt_toolkit            3.0.48
> psutil                    6.0.0
> pure_eval                 0.2.3
> pyarrow                   17.0.0
> pycodestyle               2.12.1
> pycparser                 2.22
> pyct                      0.5.0
> pyflakes                  3.2.0
> Pygments                  2.18.0
> pynndescent               0.5.13
> pyogrio                   0.9.0
> pyparsing                 3.1.4
> pyproj                    3.6.1
> PySocks                   1.7.1
> python-dateutil           2.9.0.post0
> python-json-logger        2.0.7
> pytz                      2024.2
> pywin32                   306
> pywinpty                  2.0.13
> PyYAML                    6.0.2
> pyzmq                     26.2.0
> readfcs                   1.1.8
> referencing               0.35.1
> requests                  2.32.3
> rfc3339-validator         0.1.4
> rfc3986-validator         0.1.1
> rich                      13.8.1
> roifile                   2024.9.15
> rpds-py                   0.20.0
> s3fs                      0.6.0
> scanpy                    1.10.3
> scikit-image              0.24.0
> scikit-learn              1.5.2
> scipy                     1.14.1
> seaborn                   0.13.2
> Send2Trash                1.8.3
> session_info              1.0.0
> setuptools                75.1.0
> shapely                   2.0.6
> shellingham               1.5.4
> six                       1.16.0
> slicerator                1.1.0
> sniffio                   1.3.1
> sopa                      1.1.5
> sortedcontainers          2.4.0
> soupsieve                 2.6
> spatial_image             1.1.0
> spatialdata               0.2.3
> spatialdata-io            0.1.5
> stack-data                0.6.3
> statsmodels               0.14.3
> stdlib-list               0.10.0
> sympy                     1.13.3
> tblib                     3.0.0
> terminado                 0.18.1
> threadpoolctl             3.5.0
> tifffile                  2024.9.20
> tinycss2                  1.3.0
> toml                      0.10.2
> tomli                     2.0.1
> toolz                     0.12.1
> torch                     2.4.1
> tornado                   6.4.1
> tqdm                      4.66.5
> traitlets                 5.14.3
> typer                     0.12.5
> types-python-dateutil     2.9.0.20240906
> typing_extensions         4.12.2
> typing-utils              0.1.0
> tzdata                    2024.2
> umap-learn                0.5.6
> uri-template              1.3.0
> urllib3                   2.2.3
> uv                        0.4.16
> wcwidth                   0.2.13
> webcolors                 24.8.0
> webencodings              0.5.1
> websocket-client          1.8.0
> wheel                     0.44.0
> widgetsnbextension        4.0.13
> win-inet-pton             1.1.0
> wrapt                     1.16.0
> xarray                    2024.9.0
> xarray-dataclasses        1.8.0
> xarray-datatree           0.0.14
> xarray-schema             0.0.3
> xarray-spatial            0.4.0
> yarl                      1.12.1
> zarr                      2.18.3
> zict                      3.0.0
> zipp                      3.20.2
> zstandard                 0.23.0
</details>

**Comment by quentinblampey** [comment]
Hello @lguerard, did you update anything recently, apart from spatialdata?
It may be an issue with some dependencies indeed, can you try to create a new environment from scratch?

**Comment by lguerard** [comment]
I did that multiple times and still the issue.
I did not update spatialdata specifically, but everytime I install sopa it
fetches the latest version of different packages. I'm not saying that it's
coming from spatialdata, I tried installing an older version and it also
fails. I have no idea...
Is it working fine for you through notebook or also crashing?

Le jeu. 26 sept. 2024, 15:47, Quentin Blampey ***@***.***> a
écrit :

**Comment by quentinblampey** [comment]
For me it's not crashing, I have never experienced this issue yet...
I'll let you know if it happens at some point, I can also try to re-install a new environment to see if it happens

**Comment by lguerard** [comment]
If you could, that would be great. That way I can try and figure out what could be the issue. I couldn't find anything in either the console used for jupyter-lab to be ran or in the jupyter logs.

I even tried with some data that worked like a week ago but it now crashes. I really am at a loss here :/

**Comment by quentinblampey** [comment]
I run the following command lines, and everything worked well in this new environment:

```sh
conda create -n sopa python=3.10 -y
conda activate sopa
pip install "sopa[cellpose]" ipykernel
```

Sorry, I don't know what is going wrong here. Do you have the same issue outside of Jupyter?

**Comment by lguerard** [comment]
I haven't yet tested outside of Jupyter. How would you do that, just by going pure python ?

I just tried and still the issue...
```
conda create -p S:\anaconda_envs\sopa
conda install python=3.10 -c conda-forge
pip install uv # I usually use uv as it's way faster
uv pip install sopa[cellpose,baysor] jupyter ipywidgets p_tqdm # I need some extra packages for my notebook
```

**Comment by quentinblampey** [comment]
Yes I meant in a Python script for example (just copy paste the lines of code into a `.py` file and execute it).

Does it happen also locally or on other computers?

I'll be on vacation for two weeks, I'll answer you in mid-October!

**Comment by lguerard** [comment]
Hi again,

I have now tried running it as a python script and it's not working but not giving any errors:
Here's the part of the code 
```python
print("Segmentation")
segmentation.write_patch_cells(cellpose_temp_dir, 1)
print("After")
```

It prints `"Segmentation"` but doesn't create the subfolder for the cell segmentation and doesn't print `"After"` somehow...

**Comment by lguerard** [comment]
Issue seems fixed when reinstalling correct version of `cudatoolkit` and `pytorch` somehow. Sorry about this !

**Comment by quentinblampey** [comment]
Great, good to hear that it fixed it!
